### PR TITLE
Handle deleted items

### PR DESCRIPTION
**Pull request by aelkiss**
- record responds to `deleted?`

- clean up rspec setup - enable spec_helper, load application by default
  - so we can run individual specs

- remove extranous pry requires

### CODE REVIEW

**Comment by aelkiss** [comment]
@billdueber  These all work now whereas before (and with staging.oai.kubernetes.hathitrust.org which has what's on main) they don't.

https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&metadata_prefix=oai_dc&set=hathitrust:pd
https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&metadata_prefix=oai_dc&set=hathitrust:pdus
https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&metadata_prefix=oai_dc

https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&metadata_prefix=marc21&set=hathitrust:pd
https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&metadata_prefix=marc21&set=hathitrust:pdus
https://testing.oai.kubernetes.hathitrust.org/oai?verb=ListRecords&metadata_prefix=marc21

The contents of the subsets isn't correct in terms of what rights it's looking for in individual items, but that's for another day.

**Review by moseshll — APPROVED** [review]
Looks good here. Including cleanup of a couple "little messes" (e.g., `require "pry"`) I prolly made.

File: lib/oai_solr/dublin_core.rb [hunk 1]
```diff
@@ -18,29 +17,31 @@ def encode _, record
       xml.target!
     end
 
-    def dublin_core_hash
+    private
+
+    def dublin_core_hash(record)
```
Comment (billdueber) [inline lib/oai_solr/dublin_core.rb:22]: A couple thoughts here:
 * Is this method actually returning the completed hash? I think it'll just return the list of ISBNs (or `nil` if there aren't any)
  * Catalog documents aren't guaranteed to have most of these  fields -- might want to guard against adding empty arrays  by returning `dc.reject{|_k, v| [nil, []].include? v}`

**Review by billdueber — CHANGES_REQUESTED** [review]

**Comment by aelkiss** [comment]
Should subclass `OAI::Record` to include `OAI::DeletedRecord` and have them know what to do in terms of output to MARC and DC.

File: lib/oai_solr/dublin_core.rb (hunk 1 above)
Comment (aelkiss) [reply lib/oai_solr/dublin_core.rb:23]: @billdueber Added tests to address both these issues and added the suggested `.reject`

**Comment by aelkiss** [comment]
Test passed for me locally I think because I didn't have the latest ghcr.io/hathitrust/catalog-solr-sample image locally

**Comment by aelkiss** [comment]
Tests failed again in GHA. Maddening. @billdueber @moseshll Do they pass for you locally?

**Comment by moseshll** [comment]
@aelkiss I can't get this branch to fail locally.

**Comment by aelkiss** [comment]
Argh. What is going on??

**Comment by aelkiss** [comment]
Looking at the failures with fresh eyes I suspect that the issue has to do with validating against `http://www.loc.gov/standards/marcxml/schema/MARC21slim.xsd` and in particular maybe what's going on is that it's pulling it down over and over in the tests & getting throttled or something. I will see what I can do.

**Comment by aelkiss** [comment]
@moseshll  See the latest commit. Does using `validateCache` on x86 and falling back to `StdInParse` on arm seem like a problem? Building multiarch feed seems like more rigmarole than I want just to get an ARM version of validateCache. Suppose we could just build an arm binary and bake it in to the repo? Ugh. Too much yak shaving for me...

**Comment by aelkiss** [comment]
Thoughts about how to address the ARM issue:

- Only run tests with validation on x86?
- Use jruby for tests for multi-schema validation? This seems better to me...

**Comment by aelkiss** [comment]
Created DEV-509 as a workaround for this.

**Pull request merged**